### PR TITLE
Update Release Notes for oneDPL 2022.4.0

### PR DESCRIPTION
**Pull request by ValentinaKats**
Update Release Notes for oneDPL 2022.4.0
Signed-off-by: Valentina Kats valentina.kats@intel.com

### CODE REVIEW

File: documentation/release_notes.rst [hunk 1]
```diff
@@ -8,6 +8,26 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+  They can be parameterized by a number of elements processed by a work-item,
+  number of work-items in a work-group to achieve the best performance depending on
+  input characteristics such as a number of elements and their type.
+- Added new ``transform_if`` algorithm for conditionally applying a transform based on
+  a predicate to the input sequence(s). Overloads exist for a single input sequence with
+  a unary transform and unary predicate and for a pair of input sequences with
+  a binary transform and binary predicate.
+- Improved performance of most of algorithms used with DPC++ execution policies,
+  when a program is built with oneAPI DPC++ compiler, by applying optimizations which
+  are already effective with Intel oneAPI DPC++/C++ Compiler.
```
Comment (akukanov) [inline documentation/release_notes.rst:27]:
```suggestion
- Optimizations used with Intel oneAPI DPC++/C++ Compiler are expanded to the open source oneAPI DPC++ compiler.
```

File: documentation/release_notes.rst [hunk 2]
```diff
@@ -8,6 +8,26 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+------------
+- Added highly configurable ``radix_sort`` and ``radix_sort_by_key`` algorithms
+  residing in ``oneapi::dpl::experimental::kt::esimd`` namespace, which work with Intel® Data Center GPU Max Series.
+  They can be parameterized by a number of elements processed by a work-item,
+  number of work-items in a work-group to achieve the best performance depending on
+  input characteristics such as a number of elements and their type.
+- Added new ``transform_if`` algorithm for conditionally applying a transform based on
+  a predicate to the input sequence(s). Overloads exist for a single input sequence with
+  a unary transform and unary predicate and for a pair of input sequences with
+  a binary transform and binary predicate.
```
Comment (akukanov) [inline documentation/release_notes.rst:24]:
```suggestion
- Added new ``transform_if`` algorithm for applying a transform function conditionally
  based on a predicate, with overloads provided for one and two input sequences
  that use correspondingly unary and binary operations and predicates.
```

File: documentation/release_notes.rst [hunk 3]
```diff
@@ -8,6 +8,26 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
...
+New in 2022.4.0
+===============
+
+New Features
+------------
+- Added highly configurable ``radix_sort`` and ``radix_sort_by_key`` algorithms
+  residing in ``oneapi::dpl::experimental::kt::esimd`` namespace, which work with Intel® Data Center GPU Max Series.
+  They can be parameterized by a number of elements processed by a work-item,
+  number of work-items in a work-group to achieve the best performance depending on
+  input characteristics such as a number of elements and their type.
```
Comment (akukanov) [inline documentation/release_notes.rst:20]:
```suggestion
- Added experimental ``radix_sort`` and ``radix_sort_by_key`` algorithms residing in
  the ``oneapi::dpl::experimental::kt::esimd`` namespace. These algorithms are first
  in the family of _kernel templates_ that allow configuring a variety of parameters
  including the number of elements to process by a work item, and the size of a workgroup.
  The algorithms only work with Intel® Data Center GPU Max Series.
```

File: documentation/release_notes.rst (hunk 2 above)
Comment (danhoeflinger) [reply documentation/release_notes.rst:24]: Thanks @akukanov , I think this is a more clear and concise way of putting it.

**Review by akukanov — APPROVED** [review]

**Comment by dmitriy-sobolev** [comment]
The previous PRs for release notes ([2022.3](https://github.com/oneapi-src/oneDPL/pull/1180), [2022.2](https://github.com/oneapi-src/oneDPL/pull/988)) updated "Known Issues and Limitations" -> New in This Release" section. I guess this PR should update it too.

**Pull request merged**